mapsforge-map/src/main/java/org/mapsforge/map/layer/renderer/DatabaseRenderer.java

```diff
@@ -83,7 +84,7 @@ public DatabaseRenderer(MapDataStore mapDataStore, GraphicFactory graphicFactory
      *
      * @param rendererJob the job that should be executed.
      */
-    public TileBitmap executeJob(RendererJob rendererJob) {
+    public synchronized TileBitmap executeJob(RendererJob rendererJob) {
```

Comment:
@Sublimis 

Why we need the `synchronized` here?

Reply:
> @Sublimis
> 
> Why we need the `synchronized` here?

`Canvas` (used in the `CanvasRasterer` throughout the `executeJob()` method) should not be considered thread-safe, at least not on Android. `Canvas` thread-safety assumption was also the reason for one of the artifacts mentioned above, specifically the one in the first image.

Reply:
I wonder if this affects the multi threading vector rendering (not hill shading).
(as we have not seen problems there without the `synchronized` so far)

Reply:
You mean performance? Performance is probably the only thing that could possibly be affected by synchronizing this method. Haven't noticed any decline in performance though, but will try to make some more objective testing. Can you suggest some especially complex scene that takes notable amount of time to render?

Reply:
Note: I mean the normal map rendering **without** hill shading, i.e. test only the vector maps.

Very populated areas like London at zoom levels where much data is processed /rendered,
with 1 thread we wait the tiles to render, while the (default) multi threading is much faster.

Reply:
> Very populated areas like London at zoom levels where much data is processed /rendered, with 1 thread we wait the tiles to render, while the (default) multi threading is much faster.

You mean with the current synchronized paradigm there is a noticeable wait that was not present before?
Ok, will let you know after looking into that.

Reply:
Exactly, I mean the current code in `master` branch with and without this `synchronized`,
if it affects the normal map rendering without hill shading when multi threading is enabled.

Reply:
Done and fixed. Thanks for pointing this out. The cause of artifacts like in the first image was pinpointed to be the concurrent access/modification of hill shading bitmaps through Canvas in very distant code sites, so a proper synchronization of minimum granulation was applied there. Will put the code (containing some other exciting improvements) to our fork tomorrow, so you can check it out before we make a PR.

Reply:
Sorry for the delay. You can see our latest improvements here: https://github.com/Sublimis/mapsforge/pull/14. The commit is quite intricate, please take your time and ask if anything is unclear. We're quite happy with everything, this nicely rounds off all the work done on hill shading so far, and raises it to another level.

Reply:
@Sublimis Thanks, there are too many changes in the whole library core classes just for hill shading.

Why do we need them?
The description only mentions the hill shading improvements.

Documentation is needed per change and class in the pull request
or multiple steps and pull requests to understand all of them.

The many users of the library are usually interested in the normal vector map rendering.
And if any core changes have positive or negative effect of the normal map performance.

Can you give more details about all other changes?

Also please avoid moving existing methods in other line, it makes review more difficult.
Only new or changed code should exist in commits where there is functionality change.